test/distributed/org/apache/cassandra/distributed/test/sai/features/FeaturesVersionSupportTester.java

```diff
@@ -0,0 +1,219 @@
+    {
+        cluster.schemaChange(withKeyspace("CREATE TABLE %s.q_analyzer (k int PRIMARY KEY, v text)"));
+
+        ICoordinator coordinator = cluster.coordinator(1);
+        coordinator.execute(withKeyspace("INSERT INTO %s.q_analyzer (k, v) VALUES (1, 'astra quick fox')"), ALL);
+        coordinator.execute(withKeyspace("INSERT INTO %s.q_analyzer (k, v) VALUES (2, 'astra2 quick fox')"), ALL);
+        coordinator.execute(withKeyspace("INSERT INTO %s.q_analyzer (k, v) VALUES (3, 'astra3 quick foxes')"), ALL);
+        cluster.forEach(node -> node.flush(KEYSPACE));
+
+        cluster.schemaChange(withKeyspace("CREATE CUSTOM INDEX q_analyzer_idx ON %s.q_analyzer(v) USING 'StorageAttachedIndex' WITH OPTIONS = {" +
+                                          "'index_analyzer': '{" +
+                                          "  \"tokenizer\" : { \"name\" : \"whitespace\", \"args\" : {} }," +
+                                          "  \"filters\" : [ { \"name\" : \"lowercase\", \"args\": {} }, " +
+                                          "                  { \"name\" : \"edgengram\", \"args\": { \"minGramSize\":\"1\", \"maxGramSize\":\"30\" } }]," +
+                                          "  \"charFilters\" : []}', " +
+                                          "'query_analyzer': '{" +
+                                          "  \"tokenizer\" : { \"name\" : \"whitespace\", \"args\" : {} }," +
+                                          "  \"filters\" : [ {\"name\" : \"lowercase\",\"args\": {}} ]}'}"));
+        SAIUtil.waitForIndexQueryable(cluster, KEYSPACE, "q_analyzer_idx");
+
+        Assertions.assertThat(coordinator.execute(withKeyspace("SELECT k FROM %s.q_analyzer WHERE v : 'ast'"), ONE))
+                  .hasNumberOfRows(3);
+        Assertions.assertThat(coordinator.execute(withKeyspace("SELECT k FROM %s.q_analyzer WHERE v : 'astra'"), ONE))
+                  .hasNumberOfRows(3);
+        Assertions.assertThat(coordinator.execute(withKeyspace("SELECT k FROM %s.q_analyzer WHERE v : 'astra2'"), ONE))
+                  .hasNumberOfRows(1);
+        Assertions.assertThat(coordinator.execute(withKeyspace("SELECT k FROM %s.q_analyzer WHERE v : 'fox'"), ONE))
+                  .hasNumberOfRows(3);
+        Assertions.assertThat(coordinator.execute(withKeyspace("SELECT k FROM %s.q_analyzer WHERE v : 'foxes'"), ONE))
+                  .hasNumberOfRows(1);
+    }
+
+    /**
+     * Injection to set the SAI on-disk version on the first one to the most recent one, and the tested version to the second node.
+     */
+    public static class BB
+    {
+        @SuppressWarnings("resource")
+        public static void install(ClassLoader classLoader, int node, Version version)
+        {
+            new ByteBuddy().rebase(Version.class)
+                           .method(named("latestVersion"))
+                           .intercept(FixedValue.value(node == 1 ? Version.EC.toString() : version.toString()))
```

Comment:
we will need to remember to change this test when we will have a new "latest" version

what about adding a constan tin Version (dependent on the code and not configurable by system property like the current 'latest' version)